.github/workflows/test.yml

```
@@ -0,0 +1,29 @@
+name: Test
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: true
+      matrix:
+        node-version: [12.x, 14.x, 16.x, 18.x]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - run: npm install
+
+      # GitHub actions do not use a TTY, which is required for util.inspect to
+      # output in color, so use script to force a TTY.
+      - run: script --return --quiet -c "npm test" /dev/null
```

I'm using `shell` prop in the CI steps to work around no TTY issue and keep the `run` prop clean:

https://explainshell.com/explain?cmd=script+-qec

```suggestion
      - run: npm test
        # Workaround for No TTY issue
        shell: 'script -qec "bash {0}"'
```